sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.aest.controller.ts

```diff
@@ -105,14 +116,57 @@ export class ApplicationOfferingChangeRequestAESTController extends BaseControll
    * Approves or declines an application offering change request.
    * @param applicationOfferingChangeRequestId application offering change request id.
    * @param payload information to update the application offering change request.
+   * @param userToken user approving or declining the application offering change request.
    */
+  @Roles(Role.InstitutionApproveDeclineApplicationOfferingChangeRequest)
   @Patch(":applicationOfferingChangeRequestId")
+  @ApiNotFoundResponse({
+    description:
+      "Application offering change not found or not in valid status to be updated.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description: "Invalid application offering change status",
+  })
   async updateApplicationOfferingChangeRequest(
     @Param("applicationOfferingChangeRequestId", ParseIntPipe)
     applicationOfferingChangeRequestId: number,
     @Body()
     payload: ApplicationOfferingChangeAssessmentAPIInDTO,
+    @UserToken() userToken: IUserToken,
   ): Promise<void> {
-    // TODO: API including the addition of ministry role to be done in the next PR.
+    const applicationOfferingChangeRequest =
+      await this.applicationOfferingChangeRequestService.applicationOfferingChangeRequestExists(
+        applicationOfferingChangeRequestId,
+        {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+        },
+      );
+    if (!applicationOfferingChangeRequest) {
+      throw new NotFoundException(
+        "Application offering change not found or not in valid status to be updated.",
+      );
+    }
+    if (
+      !(
+        payload.applicationOfferingChangeRequestStatus ===
+          ApplicationOfferingChangeRequestStatus.DeclinedBySABC ||
+        payload.applicationOfferingChangeRequestStatus ===
+          ApplicationOfferingChangeRequestStatus.Approved
+      )
+    ) {
+      throw new UnprocessableEntityException(
+        "Invalid application offering change request status.",
+      );
+    }
+    await this.applicationOfferingChangeRequestService.assessApplicationOfferingChangeRequest(
+      applicationOfferingChangeRequestId,
+      payload.applicationId,
+      payload.offeringId,
+      payload.studentId,
+      payload.applicationOfferingChangeRequestStatus,
+      payload.note,
```

Comment:
IMO. you can directly pass the payload to the service

Reply:
I am following this from how it is done in the students controller for `updateApplicationOfferingChangeRequest`.

Reply:
Now the 3 properties are removed. So I think it should be ok now.